File: BattlrNet/Code/Game/AABB.cs
using System;

namespace BattlrNet.Code.Game
{
    public class AABB
    {
        public Vector Center;
        public Vector HalfSize;

        public AABB(Vector center, Vector halfSize)
        {
            Center = center;
            HalfSize = halfSize;
        }

        public bool Overlaps(AABB other)
        {
            if (Math.Abs(Center.X - other.Center.X) > this.HalfSize.X + other.HalfSize.X) return false;
            if (Math.Abs(Center.Y - other.Center.Y) > this.HalfSize.Y + other.HalfSize.Y) return false;
            return true;
        }

        public Vector GetSize()
        {
            return new Vector(HalfSize.X * 2, HalfSize.Y * 2);
        }
    }
}

File: BattlrNet/ClientApp/AABB.ts
import Vector from "./Vector";
import IPoint from "./IPoint";

export default class AABB {
    constructor(public center: Vector, public halfSize: Vector) {

    }

    public overlaps(other: AABB) {
        if (Math.abs(this.center.x - other.center.x) > this.halfSize.x + other.halfSize.x) return false;
        if (Math.abs(this.center.y - other.center.y) > this.halfSize.y + other.halfSize.y) return false;
        return true;
    }

    public getSize(): IPoint {
        return {
            x: this.halfSize.x * 2,
            y: this.halfSize.y * 2
        };
    }
}